libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -2575,19 +2608,19 @@ instance IsUniverse fn => Functions (SetFn fn) fn where
       constrained $ \x ->
         unsafeExists $ \x' ->
           assert (x ==. singleton_ x') <> toPreds x' ts
-    Size ->
-      constrained $ \x ->
-        unsafeExists $ \x' ->
-          assert (x ==. size_ x') <> toPreds x' ts
 
 -- List -------------------------------------------------------------------
 
-data ListSpec fn a = ListSpec [a] (Spec fn Int) (Spec fn a) (FoldSpec fn a)
+data ListSpec fn a = ListSpec [a] (Spec fn Size) (Spec fn a) (FoldSpec fn a)
 
 deriving instance Show (FoldSpec fn a)
 deriving instance HasSpec fn a => Show (ListSpec fn a)
 
 instance HasSpec fn a => HasSpec fn [a] where
+  sizeOfTypeSpec (ListSpec xs sz elemS _) =
```

Comment:
What is the specification for `sizeOfTypeSpec`?? Here we are mixing the length of the list and the _size_ of the spec for the length of the list?